GVFS/GVFS.Common/Maintenance/PostFetchStep.cs

```diff
@@ -27,6 +27,20 @@ protected override void PerformMaintenance()
                 this.Context.FileSystem.TryDeleteFile(multiPackIndexLockPath);
 
                 this.RunGitCommand((process) => process.WriteMultiPackIndex(this.Context.Enlistment.GitObjectsRoot));
+
+                GitProcess.Result verifyResult = this.RunGitCommand((process) => process.VerifyMultiPackIndex(this.Context.Enlistment.GitObjectsRoot));
+
+                if (verifyResult.ExitCodeIsFailure)
+                {
+                    EventMetadata metadata = this.CreateEventMetadata();
+                    metadata["MultiPackIndexVerifyOutput"] = verifyResult.Output;
+                    metadata["MultiPackIndexVerifyErrors"] = verifyResult.Errors;
+                    string multiPackIndexPath = Path.Combine(this.Context.Enlistment.GitPackRoot, "multi-pack-index");
+                    metadata["TryDeleteFileResult"] = this.Context.FileSystem.TryDeleteFile(multiPackIndexPath);
+                    activity.RelatedError(metadata, "multi-pack-index is corrupt after write. Deleting and rewriting.");
+
+                    this.RunGitCommand((process) => process.WriteMultiPackIndex(this.Context.Enlistment.GitObjectsRoot));

```

Comment:
Same comment about writing the midx without validating. This would still leave enlistments broken if it was corrupted. Deletion without rewrite would leave the enlistment functional, but slow. What are your thoughts on how we should handle this?

Reply:
What if we were a big more aggressive and had WriteMultiPackIndex do the verification / retry.  This way we can be sure the Write/Verify are always paired together.  

I agree it seems like it would be good to know if the rewrite was successful.  Would it make sense to make the Verification  a warning if we succeed on the second try?

Reply:
I guess the primary question to answer is if we think regenerating the file will fix the corruption?

Reply:
The examples we've seen all have the problem that the multi-pack-index or commit-graph files were corrupted, and the problem persisted only because we continued to trust that data when we write a new one. Deleting the file and rewriting (based on the .pack and .idx files) has always fixed the issue.

If the rewrite did not fix the issue, then we would see failures from the same enlistment multiple times in a row. I plan to set alerts for any user hitting this even once, so we would be ready to intervene if this happens repeatedly.

Reply:
Makes sense. So we expect the validate/rewrite cycle to fix the issue in all cases we have seen so far? With monitors in place, I'm happy with this pattern.